### ISSUE TITLE
Next Steps in Modularizing CI/CD

### ISSUE THREAD

**Issue by aj-stein-nist** [usnistgov/OSCAL#1470]
### User Story

As a NIST OSCAL developer, to make effective use for common tasks that are part of each CI/CD process in a more reusable for downstream repos like oscal-content.

### Goals

We need to review the initial research and spike work around https://github.com/usnistgov/oscal-content/issues/116#issuecomment-1249393553 in order to:

- [ ] Itemize which tasks each kind of job (website generation; schema generation; model reference generation; content generation) have reusable building-block tasks that are repeatable.
- [ ] Complete spike or re-review if packaged Actions, reusable (composite workflows), or no solution is possible
- [ ] Revisit these goals and implement solution or plan future work in targeted issues

### Acceptance Criteria

- [ ] All [OSCAL website](https://pages.nist.gov/OSCAL) and readme documentation affected by the changes in this issue have been updated. Changes to the OSCAL website can be made in the docs/content directory of your branch.
- [ ] A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.
- [ ] The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.

**Comment by aj-stein-nist** [comment]
Per discussion with @nikitawootten-nist, this issue is one we could potentially close, but I would prefer that we do consider making reusable actions available for our team and the benefit of others that use our CI/CD utilities due to overlapping use cases and strong alignment in project inputs and outputs like FedRAMP. So, this one stays open.

**Comment by aj-stein-nist** [comment]
I still would like to revisit this re https://github.com/usnistgov/OSCAL/issues/1470#issuecomment-1664606772 but after #116 is completed. Once that is done, it can be moved from Blocked to Needs Refinement: it is open-ended and needs to focused on specific actionable steps or broken down into an epic.
